pkgs/development/libraries/jemalloc/rust.nix

```
@@ -0,0 +1,34 @@
+{ lib
+, stdenv
+, jemalloc
+, writeText
+
+, unprefixed ? false
+}:
+
+let
+  # On some platforms the unprefixed feature will be ignored:
+  # https://github.com/tikv/jemallocator/blob/ab0676d77e81268cd09b059260c75b38dbef2d51/jemalloc-sys/src/env.rs
+  unprefixed' = unprefixed && !stdenv.hostPlatform.isMusl && !stdenv.hostPlatform.isDarwin && !stdenv.hostPlatform.isAndroid;
+
+in jemalloc.overrideAttrs (oldAttrs: {
+  configureFlags = oldAttrs.configureFlags ++ lib.optionals (!unprefixed') [
+    "--with-jemalloc-prefix=_rjem_"
+  ] ++ [
+    "--with-private-namespace=_rjem_"
+  ];
+
+  postFixup = ''
+    touch $out/rust-jemalloc
+  '';
+
+  setupHook = writeText "setup-hook.sh" ''
+    addRustJemalloc () {
+      if test -f "$1/rust-jemalloc"; then
+        export JEMALLOC_OVERRIDE="$1/lib/libjemalloc${stdenv.hostPlatform.extensions.library}"
```

I think we can maybe do

```suggestion
      export JEMALLOC_OVERRIDE="@out@/lib/libjemalloc${stdenv.hostPlatform.extensions.library}"
```

and not need the `if` or the `postFixup`?